packages/node/src/dependencies.ts

```diff
@@ -141,16 +54,53 @@ export function getClientAuthenticationWithDependencies(dependencies: {
   secureStorage?: IStorage;
   insecureStorage?: IStorage;
 }): ClientAuthentication {
-  const storage = new InMemoryStorage();
-  const secureStorage = dependencies.secureStorage || storage;
-  const insecureStorage = dependencies.insecureStorage || storage;
-
-  const authenticatorContainer = container.createChildContainer();
-  authenticatorContainer.register<IStorage>("node:secureStorage", {
-    useValue: secureStorage,
-  });
-  authenticatorContainer.register<IStorage>("node:insecureStorage", {
-    useValue: insecureStorage,
-  });
-  return authenticatorContainer.resolve(ClientAuthentication);
+  const inMemoryStorage = new InMemoryStorage();
+  const secureStorage = dependencies.secureStorage || inMemoryStorage;
+  const insecureStorage = dependencies.insecureStorage || inMemoryStorage;
+
+  const storageUtility = new StorageUtilityNode(secureStorage, insecureStorage);
+
+  const issuerConfigFetcher = new IssuerConfigFetcher(storageUtility);
+  const clientRegistrar = new ClientRegistrar(storageUtility);
+
+  const sessionInfoManager = new SessionInfoManager(storageUtility);
+
+  const tokenRefresher = new TokenRefresher(
+    storageUtility,
+    issuerConfigFetcher,
+    clientRegistrar
+  );
+
+  return new ClientAuthentication(
+    // TODO: I don't think we need an Aggregate login handler here at all, since we only use one
+    //  anyways...!
+    new AggregateLoginHandler([
+      new OidcLoginHandler(
+        storageUtility,
+        new AggregateOidcHandler([
```

Comment:
Perhaps move this one up and assign to a variable? Just for readability.
Same with the other deep nestings.